core/controllers/topics_and_skills_dashboard.py

```diff
@@ -172,8 +177,31 @@ def post(self):
 
         skill.update_explanation(
             state_domain.SubtitledHtml.from_dict(explanation_dict))
+
+        image_filenames = skill_services.get_image_filenames_from_skill(skill)
+
         skill_services.save_new_skill(self.user_id, skill)
 
+        for filename in image_filenames:
+            image = self.request.get(filename)
+            if not image:
+                logging.error(
+                    'Image not provided for file with name %s when the skill '
+                    'with id %s was created.' % (filename, skill.id))
+                raise self.InvalidInputException(
+                    'No image data provided for file with name %s' % filename)
```

Comment:
Do you need a similar "if not filename" check?

Reply:
I am iterating through the filenames obtained by going through all the skill's HTML contents, so that check is not required here, right? The list of filenames is not sent from the frontend explicitly, it is obtained from the skill itself, so a 'filename = None' case doesn't occur.

Reply:
Oh yes sorry, my bad. Please ignore. Thanks!